src/components/SlateArea/slateAreaTypes.tsx

```diff
@@ -0,0 +1,103 @@
+import type { Editor, Element } from "slate";
+// import type { WidgetProps } from "components/Widget/Widget";
+import type { ReactElement, ReactNode } from "react";
+
+export interface SlateAreaEvent {
+  target: {
+    name: string;
+    value: any;
+  };
+}
+
+/**
+ * Describes a menu button to render.
+ */
+export interface HoverMenuButton {
+  kind: "block" | "mark";
+
+  /** Slate type that will get triggered when the button is pressed, e.g. "block-quote" */
+  type: string;
+
+  /** Icon to render */
+  icon: ReactElement;
+
+  onMouseDown?: (editor: Editor) => void;
+
+  isActive?: (editor: Editor) => boolean;
+}
+
+export interface InlineMenuButton {
+  icon: string;
+  label: string;
+  // widgetProps?: Partial<WidgetProps>;
```

Comment:
Might be best to make this an `any` for now and remove the import. We will never be able to pass those widget props in unless the widget will be part of shelley-cms as well.